dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl

```diff
@@ -8,18 +8,20 @@ FROM {{ .buildFrom }} AS home
 
 COPY beat {{ $beatHome }}
 
-RUN mkdir -p {{ $beatHome }}/data {{ $beatHome }}/logs && \
+RUN mkdir -p {{ $beatHome }}/data {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/logs && \
     chown -R root:root {{ $beatHome }} && \
     find {{ $beatHome }} -type d -exec chmod 0750 {} \; && \
     find {{ $beatHome }} -type f -exec chmod 0640 {} \; && \
-    chmod 0750 {{ $beatBinary }} && \
+    rm {{ $beatBinary }} && \
+    ln -s {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/elastic-agent {{ $beatBinary }} && \
+    chmod 0750 {{ $beatHome }}/data/elastic-agent-*/elastic-agent && \
 {{- if .linux_capabilities }}
     setcap {{ .linux_capabilities }} {{ $beatBinary }} && \
 {{- end }}
 {{- range $i, $modulesd := .ModulesDirs }}
     chmod 0770 {{ $beatHome}}/{{ $modulesd }} && \
 {{- end }}
-    chmod 0770 {{ $beatHome }}/data {{ $beatHome }}/logs
+    chmod 0770 {{ $beatHome }}/data {{ $beatHome }}/data/elastic-agent-{{ commit_short }}/logs
```

Comment:
If it is an officially released version like `7.9.1`, will it still contain the commit hash?

Reply:
yes this is to support snapshots, we can change function to include version or do some conditional formats

Reply:
For someone only running production builds, it will probably be unexpected to see here a hash instead of a specific version. Would be nice if we have a nice switch between the two. But then we have the problem how to separate 2 release candidates 🤔

Reply:
I would prefer we have a single way, having special case would just increase complexity IMHO.
Also, this is a good thing we could allow people to use preview build?

Reply:
I wonder how debugging is going to work. You upgrade from 7.9.1 to 7.10.0 and things go wrong. You now need to figure out which hash 7.9.1. and 7.10.0 was know which directories to look into.

Perhaps the solution here is more symlinks with the exact version number. If two 7.9.1 exists, it only links to the most recent one.

Reply:
@ruflin The scenario you are describing will only be confusing if you upgrade to nightly snapshots. In the normal production environment you should only have something like this:


7.9.0-abcd
7.9.1-abcd2
7.9.2-bcd
7.10.0-huhuh

Now, concerning the rollback in debugging while using multiples snapshots, I believe you are right it might be confusing.

@michalpristas How do you plan to solve the following issue, note this might be an edge case.

1. I run 7.10-{hash1} (snapshot generated on monday)
2. I try to upgrade to 7.10-{hash2} It fails.
3. Rollback to 7.10-{hash1}

We could probable solve the situation with symlinks or traces on disk.

Reply:
@ph I believe there is only ever 1 previous version in the data directory during upgrade. Once upgrade is successful, the new Agent will delete the other version directory.

This makes it easy to know which is the previous version, its just not the current version.

Reply:
Yes, forgot about that. so its a non issue :)